client/src/components/AuthorizationCodeCallback.js

```diff
@@ -0,0 +1,65 @@
+import { useEffect, useRef, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+const AuthorizationCodeCallback = () => {
+    const [response, setResponse] = useState(null);
+    const [error, setError] = useState("");
+    const hasFetchedRef = useRef(false);
+
+    const navigate = useNavigate();
+
+    useEffect(() => {
+      const urlParams = new URLSearchParams(window.location.search);
+      const code = urlParams.get('code');
+
+      if (hasFetchedRef.current) {
```

Comment:
Why do we need the ref here?

Reply:
React strict mode was forcing double re render, calling the enpoint twice and returning an error. Could also remove react strict mode.

Reply:
Or if there's some other cleaner way of solving it im open.

Reply:
Cool 

Worth adding that as a comment above the condition

Reply:
Added!